assembly/src/assembler/context.rs

```diff
@@ -239,12 +239,14 @@ impl AssemblyContext {
     /// - If this module is not an executable module.
     /// - If any of the procedures in the module's callset cannot be found in the specified
     ///   procedure cache or the local procedure set of the module.
-    pub fn into_cb_table(self, proc_cache: &BTreeMap<ProcedureId, Procedure>) -> CodeBlockTable {
-        assert!(!self.module_stack.is_empty(), "no modules");
-        assert_eq!(self.module_stack.len(), 1, "more than one final module");
+    pub fn into_cb_table(mut self, proc_cache: &ProcedureCache) -> CodeBlockTable {
+        // get the last module off the module stack
+        let mut main_module_context = self.module_stack.pop().expect("no modules");
+        assert!(self.module_stack.is_empty(), "executable not last module");
```

Comment:
usually its better both for the compiler & the users to put the assertions in the immediate beginning of the function. we could, then, do something like `assert!(self.module_stack.len() > 1, "executable not last module")`

Reply:
Agreed - though in this case, I first want to check if there are any modules on the stack at all (which is done with `pop().expect(...)`). We could re-write it to do all the checks first, but I'm not sure it is worth it.